python/ray/tune/suggest/suggestion.py

```diff
@@ -172,8 +173,10 @@ def __init__(self, searcher, max_concurrent):
     def suggest(self, trial_id):
         if len(self.live_trials) >= self.max_concurrent:
             return
-        self.live_trials.add(trial_id)
-        return self.searcher.suggest(trial_id)
+        suggestion = self.searcher.suggest(trial_id)
+        if suggestion:
```

Comment:
maybe this? guessing what you have is harmless though
```suggestion
        if suggestion not in (None, Suggestor.FINISHED):
```